pkg/asset/agent/installconfig.go

```diff
@@ -170,7 +170,12 @@ func (a *OptionalInstallConfig) validateSNOConfiguration(installConfig *types.In
 
 		var workers int
 		for _, worker := range installConfig.Compute {
-			workers = workers + int(*worker.Replicas)
+			if worker.Replicas == nil {
+				fieldPath = field.NewPath("Compute", "Replicas")
```

Comment:
Isn't the check at line [166](https://github.com/openshift/installer/pull/6433/files#diff-ae1557c35cb29b1eb4fd7d286f859053055f437cb2177c8b10285a098aad5246L166) already being executed in such case?

Reply:
It will work when we don't have the `compute` in the file but if we have `compute` field and does not have a `replicas` under it then it wont work. Below is the example:

```
compute:
- architecture: amd64
  hyperthreading: Enabled
  name: worker
controlPlane:
  architecture: amd64
  hyperthreading: Enabled
  name: master
  replicas: 1
```

Reply:
Thanks @mhanss I missed the dereferencing problem initially. Even though I think that those kind of validations should be refactored into the `InstallConfig` asset, the fix looks reasonable.
It would be nice to add such test case [here](https://github.com/openshift/installer/blob/869b004e527207be564710e643498d01c16b580c/pkg/asset/agent/installconfig_test.go#L18) to ensure a proper coverage

Reply:
@andfasano I have added the test case, thanks.